### PR TITLE
Enables single-page guides in navigation

### PR DESCRIPTION
**Pull request by shawyu-okta**
## Description:
- **What's changed?** 
- **Is this PR related to a Monolith release?** 

### Resolves:

* [OKTA-355337](https://oktainc.atlassian.net/browse/OKTA-355337)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/a-single-page-guide/index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+title: My Single-page Guide
+meta:
+  - name: description
+    content: Content description
+sections:
+  - main
+---
```
Comment (shawyu-okta) [inline packages/@okta/vuepress-site/docs/guides/a-single-page-guide/index.md:8]: I'll be removing this example and the associated `.md` files before merging.

File: packages/@okta/vuepress-theme-prose/components/Sidebar.vue [hunk 2]
```diff
@@ -104,23 +104,33 @@ export default {
...
-                      ? framework
-                      : guide.mainFramework
-                  )
+
+            if (Array.isArray(guide?.sections)) {
+              const [firstSection] = guide.sections;
+
+              // Special value for guide that only has one section and should be
+              // linked at the parent
+              if (firstSection.name === 'main') {
```
Comment (TiffanyNg-okta) [inline packages/@okta/vuepress-theme-prose/components/Sidebar.vue:119]: Is it a pre-req for new single page guides to then have a section named "main"? Should we also check if the number of sections === 1?

File: packages/@okta/vuepress-site/.vuepress/scripts/build-guides-info.js [hunk 3]
```diff
@@ -21,26 +21,35 @@ const getFrameworksFor = path => {
 
 const guideInfo = {};
 
+// Load frontmatter yaml to JS from root file index page
```
Comment (TiffanyNg-okta) [inline packages/@okta/vuepress-site/.vuepress/scripts/build-guides-info.js:24]: +1 on the comments

File: packages/@okta/vuepress-theme-prose/components/Sidebar.vue (hunk 2 above)
Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/components/Sidebar.vue:119]: > Is it a pre-req for new single page guides to then have a section named "main"?
For now, yes. The docs teams is OK with a convention of always naming it `main`. 

> Should we also check if the number of sections === 1?
I was thinking about only using the heuristic of `if sections === 1` to always denote a single page guide. 

I'll add this check too.

If down the line we want to remove the naming convention and only use the `sections === 1` heuristic then it's simple to do so and allow names other than `main.

File: packages/@okta/vuepress-site/.vuepress/scripts/build-guides-info.js (hunk 3 above)
Comment (shawyu-okta) [reply packages/@okta/vuepress-site/.vuepress/scripts/build-guides-info.js:24]: I had to write all of them while I was originally trying to understand the code so I figured I should just leave them 😄

Comment (volodymyrmakarchuk-okta) [reply packages/@okta/vuepress-site/.vuepress/scripts/build-guides-info.js:24]: +1, also I was thinking about refactoring (or splitting) `getGuides` method. Right now it looks a bit complicated

**Review by volodymyrmakarchuk-okta — APPROVED** [review]

File: packages/@okta/vuepress-theme-prose/components/Sidebar.vue (hunk 2 above)
Comment (francoislehoux-okta) [reply packages/@okta/vuepress-theme-prose/components/Sidebar.vue:119]: Wouldn't a value like `single-page` be a bit more explicit down the road?

Comment (shawyu-okta) [reply packages/@okta/vuepress-theme-prose/components/Sidebar.vue]: > Wouldn't a value like single-page be a bit more explicit down the road?

Part of the issue is that the section name appears in the URL of the page, so we're trying to keep is semi-innocuous

**Review by francoislehoux-okta — APPROVED** [review]

**Review by TiffanyNg-okta — APPROVED** [review]

**Pull request merged**